crates/adapters/src/transport/secret_resolver.rs

```diff
@@ -0,0 +1,260 @@
+use regex::Regex;
+use std::fmt;
+use std::fmt::{Debug, Display, Formatter};
+use std::fs;
+
+use log::{debug, warn};
+use pipeline_types::secret_ref::SimpleStringOrSecretRef;
+
+/// Enumeration which holds a simple string or a resolved secret's string.
+///
+/// This enumeration exists such that a resolved secret's string is not shown
+/// upon formatting (irrespective if `Display` or `Debug`). This is preferable
+/// over using a simple `String`, which would show its content upon formatting.
+///
+/// In addition, a warning is emitted when a secret is attempted to be
+/// formatted.
+#[derive(Clone, PartialEq, Eq)]
+pub enum SimpleOrSecretString {
+    Simple(String),
+    Secret(String),
+}
+
+impl Display for SimpleOrSecretString {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        match self {
+            SimpleOrSecretString::Simple(simple_string) => {
+                write!(f, "Simple({})", simple_string)
+            }
+            SimpleOrSecretString::Secret(_) => {
+                warn!(
+                    "Resolved secret is requested to be formatted. \
```

Comment:
I don't think we need the warning. We might just want to display entire types that have some secret fields, and so it is benign to call Display.

Reply:
True, by shielding them, the formatting of them has become benign.

Reply:
Removed warning.